modelon/impact/client/experiment_definition.py

```diff
@@ -217,20 +234,13 @@ def __init__(
         _assert_successful_compilation(fmu)
         self.fmu = fmu
         self.custom_function = custom_function
-        self._solver_options = (
-            dict(custom_function.get_solver_options())
-            if solver_options is None
-            else dict(solver_options)
-            if isinstance(solver_options, ExecutionOptions)
-            else solver_options
+        self._solver_options = _parse_options(
+            dict(custom_function.get_solver_options()), solver_options
         )
```

Comment:
I think the first argument should just be the funtion which can be used for getting the default `custom_function.get_solver_options`. The rest can be done in the function. The main reason for this is so to only call the server asking for options if we are going to use them. Now we will do the call regardless.

For when handling the case where you want an empty by default you can send in `dict`.